File: bits_simulators_portal_frontend/src/app/models/EmvTag.ts
export class EmvTag {
    tag: string;
    name: string;
    shortName: string;
    format: string;
    length: string;
    source: string;
    description: string;
    template: string;

    constructor(
        tag: string,
        name: string,
        shortName: string,
        format: string,
        length: string,
        source: string,
        description: string,
        template: string
    ) {
        this.tag = tag;
        this.name = name;
        this.shortName = shortName;
        this.format = format;
        this.length = length;
        this.source = source;
        this.description = description;
        this.template = template;
    }
}

File: bits_simulators_portal_backend/src/main/java/com/simulator/smartICC/models/EmvTag.java
package com.simulator.smartICC.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "EMVTAG")
public class EmvTag {
    @Id
    //@GeneratedValue(strategy = GenerationType.IDENTITY)

    @Column(name = "TAG")
    private String tag;
    @Column(name = "NAME")
    private String name;
    @Column(name = "SHORTNAME")
    private String shortName;
    @Column(name = "FORMAT")
    private String format;
    @Column(name = "LENGTH")
    private String length;
    @Column(name = "SOURCE")
    private String source;
    @Column(name = "DESCRIPTION", length = 500)
    private String description;
    @Column(name = "TEMPLATE")
    private String template;
}
